lib/algos/list/Extension.js

```diff
@@ -23,6 +23,32 @@ class Extension {
         this.keys = 0;
     }
 
+    /**
+     * Filters-out non-requested optional fields from the value. This function
+     * shall be applied on any value that is to be returned as part of the
+     * result of a listing extension.
+     *
+     * @param {Object} value - The JSON value of a listing item
+     *
+     * @return {Object} The value without the filtered-out fields.
+     */
+    filterContent(value) {
+        let ret = undefined;
+        try {
+            ret = JSON.parse(value);
```

Comment:
As an improvement attempt to limit the CPU cost I would prefer to have a heuristic condition on the blob size before executing this, so the location field will be trimmed only if the blob is really big. Then we pay the serialize/deserialize cost only when we're certain to gain a significant number of bytes in the result (only if our assumption that the "location" field takes up the bulk of the blob bytes, which in theory should be the case).

This should amortize the cost only to memory-greedy listings and not have an impact on the vast majority of listings, considering the blob size check to be much cheaper than the serialization cycle.

Reply:
Good point. LGTM, that was one of my worries